### PR TITLE
Revert "Revert "[Core] minor refactor for usability of runtime_env (#15965)" (#15985)"

### PR DESCRIPTION
**Pull request by ijrsvt**
This reverts commit ec522a2d46d0a808b8af37140d5fb39c61e677bb.

## Related issue number

The main change here is that the pypa wheel builder went from creating wheels in the format of 
`ray-2.0.0.dev0-cp37-cp37m-linux_x86_64.whl` to `ray-2.0.0.dev0-cp37-cp37m-manylinux2014_x86_64.whl`.

The problem is that we've seen cases where both variants (just `linux` and `manylinux2014` are output in different runs). The new logic is to replace on **-linux** (proceeding dash) with **-manylinux2014**, meaning that output in either format will be canonicalized into `manylinux2014`.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by architkulkarni — APPROVED** [review]

**Review by simon-mo — APPROVED** [review]

File: python/build-wheel-manylinux2014.sh [hunk 1]
```diff
@@ -95,7 +95,10 @@ done
 # hack, we should use auditwheel instead.
 for path in .whl/*.whl; do
   if [ -f "${path}" ]; then
-    mv "${path}" "${path//linux/manylinux2014}"
+    out="${path//-linux/-manylinux2014}"
+    if [ "$out" != "$path" ]; then
+        mv "${path}" "${out}"
```
Comment (AmeerHajAli) [inline python/build-wheel-manylinux2014.sh:100]: why is this necessary?

Comment (ijrsvt) [reply python/build-wheel-manylinux2014.sh:100]: Updated the PR description to explain. The reason for the  `"$out" != "$path"` is that `mv` fails if src & dst are the same location.

**Comment by ijrsvt** [comment]
Only failure is MacOS Flaky failure (in which all tests passed)!
